src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3739,15 +3739,16 @@ private void LearnFromEqualsMethod(MethodSymbol method, BoundCall node, TypeWith
                 || method.MethodKind != MethodKind.Ordinary
                 || method.ReturnType.SpecialType != SpecialType.System_Boolean
                 || (method.Name != SpecialMembers.GetDescriptor(SpecialMember.System_Object__Equals).Name
-                    && method.Name != SpecialMembers.GetDescriptor(SpecialMember.System_Object__ReferenceEquals).Name))
+                    && method.Name != SpecialMembers.GetDescriptor(SpecialMember.System_Object__ReferenceEquals).Name
+                    && !method.IsExplicitInterfaceImplementation()))
```

Comment:
>!method.IsExplicitInterfaceImplementation() [](start = 23, length = 43)

I don't think this is going to help with the name mismatch. For example, the method itself might not implement an interface, but the method it overrides does. Please add a test for a scenario like that. Perhaps we should traverse all overrides and check all of them?

Reply:
Fixed